### DISCUSSION TITLE
Run an action from name stored in DB

### DISCUSSION THREAD

**Discussion by praveenkalaiarasu** [lorisleiva/laravel-actions#198] [General]
I have my action classes' names stored in DB with full namespace.
Can I run them using this package?

Example,

```php
<?php

namespace Modules\Auth\Actions\Password;

class ResetPassword
```

is stored in DB as

```php
Model: ActionStep
Column: StepClass

Modules\Auth\Actions\Password\ResetPassword
```

in string format and currently I run them using,

```php
app($action_step['step_class'], [
	'payload' => $this->payload
]);
```
where i receive the `$payload` in the constructor and process it then. Hope my intention is clear.

**Comment by leandrodiogenes** [comment]
Hi @pravnkay .
I'm sorry, but I couldn't understand your question.

**Reply by praveenkalaiarasu** [reply]
Ok, let me try to explain more this time.
Lets assume,
1. `ActionClass` means the class which uses the `AsAction` trait.

Now, lets take the example from your docs [here](https://laravelactions.com/2.x/basic-usage.html#running-as-an-object)

I have an `ActionClass `

```php
namespace App\Authentication\Actions;

use Lorisleiva\Actions\Concerns\AsAction;

class UpdateUserPassword
{
    use AsAction;

    public function handle(User $user, string $newPassword)
    {
        $user->password = Hash::make($newPassword);
        $user->save();
    }
}
```

Now, I have this above fully name-spaced class name stored in my DB like,

`App\Authentication\Actions\UpdateUserPassword`

which I retrieve as a string. Lets say I have it stored in variable `$className`

Now, can I call this `ActionClass` using the name of the class and also pass parameters.

__________________________________________________________

If it is a non `ActionClass`, the below code works and it calls the `__construct` method of the class.
```php

app($className, [
    'payload' => $this->payload
]);

```

If it is an `ActionClass` like above, its not calling the handle function. I need to create a `__construct` method and from there I need to call `handle` method

Is there a way to do maybe like, 

```php
$className::run($payload)
```

Hope I am I clear this time.

**Reply by praveenkalaiarasu** [reply]
Ok, while writing the above reply only I thought of the last line. and I tried it and it works.
So, if the fully name-spaced class-name is in a variable (`$className`) as a string, you can actually do,

```php
$className::run($payload)
```

Thanks for being active & responding promptly!

**Comment by leandrodiogenes** [comment]
Try to run as job, so you can use dispatch method.
Sorry im not on Pc right now, but you can try to run like this.
```php
dispatch(app($className, [
    'payload' => $this->payload
]));
